lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -272,7 +272,7 @@ static FunctionType getComponentType(ComponentOp component) {
 }
 
 /// Returns the port information for the given component.
-SmallVector<ComponentPortInfo> calyx::getComponentPortInfo(Operation *op) {
+SmallVector<PortInfo> calyx::getComponentPortInfo(Operation *op) {
```

Comment:
As above + can we do `calyx::getPortInfo(ComponentOp op)`? that will remove the assert + dyn_cast in the function body, and as far as i can tell, most (all?) callsites of this function in the code already use an instance of `ComponentOp`.

Reply:
Yeah I see what you're saying. Perhaps its better in this case to just make it a member function of ComponentOp.

Reply:
Need to fix a segmentation fault, but this is the general idea. WDYT?